### PR TITLE
fix(suite): legacy component cleanup

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve

### CODE REVIEW

File: packages/components/src/components/Box/Box.tsx [hunk 1]
```diff
@@ -10,9 +11,9 @@ export interface BoxProps extends HTMLAttributes<HTMLDivElement> {
 const Wrapper = styled.div<{ state: BoxProps['state'] }>`
     display: flex;
     flex: 1;
-    border-radius: 8px;
-    padding: 16px 14px;
-    border: solid 1px ${({ theme }) => theme.STROKE_GREY};
+    border-radius: ${borders.radii.sm};
```
Comment (jvaclavik) [inline packages/components/src/components/Box/Box.tsx:14]: We don't have `8px` border in design tokens and we have this radius in multiple places. What should we do about it?

File: packages/components/src/components/Box/Box.tsx [hunk 2]
```diff
@@ -10,9 +11,9 @@ export interface BoxProps extends HTMLAttributes<HTMLDivElement> {
 const Wrapper = styled.div<{ state: BoxProps['state'] }>`
     display: flex;
     flex: 1;
-    border-radius: 8px;
-    padding: 16px 14px;
```
Comment (jvaclavik) [inline packages/components/src/components/Box/Box.tsx:14]: We don't have 14px spacing

File: packages/components/src/components/Timerange/Timerange.tsx [hunk 3]
```diff
@@ -570,24 +581,24 @@ const Calendar = styled.div`
...
     .rdrEndEdge {
-        background: ${({ theme }) => theme.BG_GREEN};
-        color: ${({ theme }) => theme.TYPE_WHITE};
-        box-shadow: ${({ theme }) => theme.BG_GREEN} 0px 0px 0px 2px;
-        border-radius: 4px;
+        background: ${({ theme }) => theme.backgroundPrimaryDefault};
+        color: ${({ theme }) => theme.textOnPrimary};
+        box-shadow: ${boxShadows.focusedLight};
+        border-radius: ${borders.radii.xs};
         z-index: ${variables.Z_INDEX.BASE};
```
Comment (jvaclavik) [inline packages/components/src/components/Timerange/Timerange.tsx]: Shouldn't we also create design tokens for z-indexes? Maybe just rename it to have it without upper case

File: packages/components/src/components/ConfirmOnDevice/ConfirmOnDevice.tsx [hunk 4]
```diff
@@ -78,29 +77,25 @@ const Right = styled(Column)``;
...
 
 const Close = styled.div`
     border-radius: 100%;
     cursor: pointer;
-    background: ${({ theme }) => theme.STROKE_GREY};
-    width: 35px;
-    height: 35px;
+    background: ${({ theme }) => theme.iconPrimaryDefault};
+    width: 36px;
+    height: 36px;
```
Comment (jvaclavik) [inline packages/components/src/components/ConfirmOnDevice/ConfirmOnDevice.tsx:95]: 36px according to design

File: packages/components/src/components/Timerange/Timerange.tsx (hunk 3 above)
Comment (jvaclavik) [reply packages/components/src/components/Timerange/Timerange.tsx]: Oh! I found `ZIndices`. Not ready yet, OK.

TIL: I didn't know that index has plural indices!

File: packages/components/src/components/Box/Box.tsx (hunk 1 above)
Comment (dahaca) [reply packages/components/src/components/Box/Box.tsx:14]: use `radii.sm` – 12px :) The UI got rounde.

File: packages/components/src/components/Box/Box.tsx (hunk 2 above)
Comment (dahaca) [reply packages/components/src/components/Box/Box.tsx:14]: use `12px` – `spacings.sm`

File: packages/components/src/components/Timerange/Timerange.tsx (hunk 3 above)
Comment (dahaca) [reply packages/components/src/components/Timerange/Timerange.tsx]: Nice :D Im thinking, for now we can copy values from `variables.Z_INDEX` to `ZIndices` as well for future use. also – easier to mass replace later.

Comment (dahaca) [reply packages/components/src/components/Timerange/Timerange.tsx]: Not necessarily as a part of this PR

Comment (jvaclavik) [reply packages/components/src/components/Timerange/Timerange.tsx]: New PR with z-indices here: https://github.com/trezor/trezor-suite/pull/9702

File: packages/suite/src/actions/suite/routerActions.ts [hunk 5]
```diff
@@ -118,18 +118,29 @@ export const goto =
...
                 setTimeout(() => dispatch(onAnchorChange(anchor)), 0);
             }
             return;
         }
-
         const newUrl = `${urlBase}${preserveParams ? history.location.hash : ''}`;
         dispatch(onLocationChange(newUrl, anchor));
 
         const route = findRouteByName(routeName);
+        console.log('__________2', newUrl, route);
```
Comment (dahaca) [inline packages/suite/src/actions/suite/routerActions.ts]: probably forgot to remove?

File: packages/suite/src/actions/suite/routerActions.ts [hunk 6]
```diff
@@ -118,18 +118,29 @@ export const goto =
         const urlBase = getPrefixedURL(getRoute(routeName, params));
 
         if (urlBase === router.url) {
+            console.log(
```
Comment (dahaca) [inline packages/suite/src/actions/suite/routerActions.ts]: probably forgot to remove?

File: packages/components/src/components/typography/Link/Link.tsx [hunk 7]
```diff
@@ -82,12 +74,7 @@ const Link = ({ icon, iconProps, ...props }: LinkProps) => {
             {props.children}
             {icon && (
                 <IconWrapper>
-                    <Icon
-                        size={props.size ? Number(A_SIZES[props.size].replace('px', '')) : 24}
-                        icon={icon}
-                        color={theme.TYPE_DARK_GREY}
-                        {...iconProps}
-                    />
+                    <Icon type={props.type} icon={icon} color={theme.iconSubdued} {...iconProps} />
```
Comment (dahaca) [inline packages/components/src/components/typography/Link/Link.tsx]: `Icon` shouldn't have `type` 🤔

File: packages/components/src/components/typography/Heading/Heading.tsx [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import styled, { css } from 'styled-components';
 
-import { FONT_WEIGHT, FONT_SIZE } from '../../../config/variables';
+import { FONT_WEIGHT } from '../../../config/variables';
+import { spacingsPx, typography } from '@trezor/theme';
 
 export interface HeadingProps {
     textAlign?: 'left' | 'center' | 'right' | 'justify';
```
Comment (dahaca) [inline packages/components/src/components/typography/Heading/Heading.tsx]: `fontWeight` is not needed

File: packages/components/src/components/typography/Heading/Heading.tsx [hunk 9]
```diff
@@ -27,36 +28,28 @@ const baseStyles = css`
     line-height: normal;
     letter-spacing: normal;
     ${(props: HeadingProps) => props.fontWeight && fontWeightStyle}
-    color: ${({ theme }) => theme.TYPE_DARK_GREY};
+    color: ${({ theme }) => theme.textDefault};
```
Comment (dahaca) [inline packages/components/src/components/typography/Heading/Heading.tsx]: this is redundant as well, I believe. Should already be in the global style somewhere

File: packages/components/src/components/typography/Heading/Heading.tsx [hunk 10]
```diff
@@ -27,36 +28,28 @@ const baseStyles = css`
     line-height: normal;
```
Comment (dahaca) [inline packages/components/src/components/typography/Heading/Heading.tsx]: line-height and spacing are already defined in `typography`, no need to have them here.

File: packages/components/src/components/typography/Heading/Heading.tsx (hunk 9 above)
Comment (dahaca) [reply packages/components/src/components/typography/Heading/Heading.tsx]: text-rendering and font- smoothing should all be in [GlobalStyles](https://github.com/trezor/trezor-suite/blob/7d23220dc5079f2433feb107fe9efac2cfad2ff9/packages/suite/src/support/suite/styles/GlobalStyle.tsx#L21), so feel free to remove

Comment (dahaca) [reply packages/components/src/components/typography/Heading/Heading.tsx]: I think everything except for `text-align` could be removed. UPD: just checked - it's not used anywhere – could be removed completely.

File: packages/components/src/components/loaders/Progress/Progress.tsx [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import styled from 'styled-components';
 
 const Wrapper = styled.div`
-    background: ${({ theme }) => theme.STROKE_GREY};
+    background: ${({ theme }) => theme.borderOnElevation0};
```
Comment (dahaca) [inline packages/components/src/components/loaders/Progress/Progress.tsx]: `backgroundNeutralSubdued`, like in [Stepper](https://github.com/trezor/trezor-suite/blob/6752b2aee2920e4f83ed066028527e46b98746c3/packages/components/src/components/loaders/Stepper/Stepper.tsx#L31)

Comment (dahaca) [inline packages/components/src/components/loaders/Progress/Progress.tsx]: Please add ` border-radius: ${borders.radii.full};` like in [Stepper](https://github.com/trezor/trezor-suite/blob/6752b2aee2920e4f83ed066028527e46b98746c3/packages/components/src/components/loaders/Stepper/Stepper.tsx#L31) :)

File: packages/suite/src/actions/suite/routerActions.ts (hunk 6 above)
Comment (jvaclavik) [reply packages/suite/src/actions/suite/routerActions.ts]: whaaat :D different PR 😂

**Review by dahaca — APPROVED** [review]
🥂

**Pull request merged**